network/transport/grpc/connection_manager_test.go

```diff
@@ -303,6 +303,47 @@ func Test_grpcConnectionManager_openOutboundStreams(t *testing.T) {
 	})
 }
 
+func Test_grpcConnectionManager_openOutboundStream(t *testing.T) {
+	protocol := &TestProtocol{}
+
+	t.Run("ok", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+
+		grpcPeer := &peer.Peer{}
+		peerInfo := transport.Peer{}
+		ctx := peer.NewContext(context.TODO(), grpcPeer)
+		nodeDID, _ := did.ParseDID("did:nuts:test")
+
+		authenticator := NewMockAuthenticator(ctrl)
+		authenticator.EXPECT().Authenticate(*nodeDID, *grpcPeer, peerInfo).Return(peerInfo, nil)
```

Comment:
I'd return a different peerInfo with nodeDID populated ("authenticated" peer), because that's provided to `setPeer`. Because now `setPeer` can be fed the unauthenticated `Peer`, and the test will still succeed.

Reply:
Good one, will fix!